lua/starfall/libs_sv/prop.lua

```diff
@@ -0,0 +1,154 @@
+	inst.data.props.count = 0
+
+	insts[inst]= nil
+end)
+
+local function propOnDestroy(propent, propdata)
+	if not propdata.props then return end
+	local prop = SF.Entities.Wrap(propent)
+	if propdata.props[prop] then
+		propdata.props[prop] = nil
+		propdata.count = propdata.count - 1
+		assert(propdata.count >= 0)
+	end
+end
+
+
+--- Updates/Checks burst constraints
+-- @class function
+-- @param instance Instance table for the burst values related to current SF Instance / Player
+-- @param noupdate False if updating the burst should be done.
+local function can_spawn(instance, noupdate)
+	if instance.data.props.burst > 0 then
+		if not noupdate then instance.data.props.burst = instance.data.props.burst - 1 end
+		return true
+	else
+		return false
+	end
+end
+
+--- Checks if the total number of props across all instances has reached the max limit.
+-- @class function
+-- @return True/False depending on if limit has been reached for SF Props
+local function max_reached()
+	local c = 0
+	for _, v in pairs( plyCount ) do
+		c = c + v
+	end
+	if c >= SF.Props.defaultquota:GetInt() then return true else return false end
+end
+
+--- Checks if the users personal limit of props has been exhausted
+-- @class function
+-- @param i Instance to use, this will relate to the player in question
+-- @return True/False depending on if the personal limit has been reached for SF Props
+local function personal_max_reached( i )
+	return plyCount[i.player] >= SF.Props.personalquota:GetInt()
+end
+
+timer.Create( "SF_Prop_BurstCounter", 1/4, 0, function()
+	for i, _ in pairs( insts ) do
+		if i.data.props.burst < SF.Props.burstrate:GetInt() or 10 then -- Should allow for dynamic changing of burst rate from the server.
+			i.data.props.burst = i.data.props.burst + 1
+		end
+	end
+end )
+
+--- Creates a prop.
+-- @server
+-- @return The prop object
+function props_library.create ( pos, ang, model, frozen )
+	SF.CheckType( pos, SF.Types[ "Vector" ] )
+	SF.CheckType( ang, SF.Types[ "Angle" ] )
+	SF.CheckType( model, "string" )
+	frozen = frozen and true or false
+
+	local pos = vunwrap( pos )
+	local ang = SF.Angles.Unwrap( ang )
+
+	local instance = SF.instance
+	if not can_spawn( instance ) then return SF.throw( "Can't spawn props that often", 2 )
+	elseif personal_max_reached( instance ) then return SF.throw( "Can't spawn props, maximum personal limit of " .. SF.Props.personalquota:GetInt() .. " has been reached", 2 )
+	elseif max_reached() then return SF.throw( "Can't spawn props, maximum limit of " .. SF.Props.defaultquota:GetInt() .. " has been reached", 2 ) end
+	if not IsValid( instance.player ) then return SF.Entities.Wrap( NULL ) end
+	if not gamemode.Call( "PlayerSpawnProp", instance.player, model ) then return SF.Entities.Wrap( NULL ) end
+
+	local propdata = instance.data.props
+	local propent = ents.Create( "prop_physics" )
+	
+	propent:CallOnRemove( "starfall_propgram_delete", propOnDestroy, propdata )
```

Comment:
Nice 'propgram' you have here. Shame if this entire PR was a copy/paste of my hologram library, just reworded slightly.